src/Cassandra/Orleans.Clustering.Cassandra/Hosting/CassandraClusteringOptions.cs

```diff
@@ -10,6 +11,30 @@ namespace Orleans.Clustering.Cassandra.Hosting;
 /// </summary>
 public class CassandraClusteringOptions
 {
+    /// <summary>
+    /// Optionally configure time-to-live behavior for the membership table row data in Cassandra itself, allowing
+    /// defunct silo cleanup even if a cluster is no longer running.
+    /// <para/>
+    /// When this is <c>true</c>, <see cref="ClusterMembershipOptions.DefunctSiloCleanupPeriod"/> must be non-null as
+    /// well (to verify that defunct cleanup is desired), and only then will the Cassandra TTL be set to
+    /// <see cref="ClusterMembershipOptions.DefunctSiloExpiration"/>.
+    /// </summary>
+    /// <remarks>
+    /// Initial implementation of https://github.com/dotnet/orleans/issues/9164 in that it only affects silo entries
+    /// that are updated with IAmAlive and will not attempt to update, for instance, the entire membership table. It
+    /// also will not affect membership tables that have already been created, since it uses the Cassandra table-level
+    /// <c>default_time_to_live</c>.
+    /// </remarks>
+    public bool UseCassandraTtl { get; set; }
+
+    internal int? GetCassandraTtlSeconds(ClusterMembershipOptions clusterMembershipOptions) =>
+        UseCassandraTtl && clusterMembershipOptions.DefunctSiloCleanupPeriod.HasValue
```

Comment:
Do we need the check for the CleanupPeriod value here? Presumably it would be valid to request the Cassandra TTL mechanism and not the periodic task.

Reply:
I suppose we don't need to, I was treating it as "if the user doesn't have `DefunctSiloCleanupPeriod` defined they probably don't want any cleanup at all, but you're right they could just choose to have the TTL-based cleanup.

Would it then be strange to continue to rely on the `DefunctSiloExpiration` in that case for the TTL?

Reply:
Sounds good, I can remove this check then and update the comments